plugins/colorschemes/base16/default.nix

```
@@ -37,94 +39,134 @@ helpers.vim-plugin.mkVimPlugin config {
-              base00 = "#16161D";
-              base01 = "#2c313c";
-              base02 = "#3e4451";
-              base03 = "#6c7891";
-              base04 = "#565c64";
-              base05 = "#abb2bf";
-              base06 = "#9a9bb3";
-              base07 = "#c5c8e6";
-              base08 = "#e06c75";
-              base09 = "#d19a66";
-              base0A = "#e5c07b";
-              base0B = "#98c379";
-              base0C = "#56b6c2";
-              base0D = "#0184bc";
-              base0E = "#c678dd";
-              base0F = "#a06949";
-            }
-          ```
+
+          You may use the name of a builtin colorscheme or an attrs that specifies the colors explicitly.
+
+          Examples of builtin themes include:
+          ${concatStrings (
+            map (e: ''
+              - "${e}"
+            '') builtinColorschemeExamples
+          )}
+
+          `:h nvim-base16-builtin-colorschemes` includes a full list of builtin themes,
+          however the [plugin's source code] may be more up to date.
+
+          [plugin's source code]: https://github.com/RRethy/base16-nvim/blob/master/lua/colors/init.lua
         '';
-        example = "edge-light";
+        example = customColorschemeExample;
       };
 
     setUpBar = mkOption {
       type = types.bool;
       default = true;
+      example = false;
       description = "Whether to set your status bar theme to 'base16'.";
     };
   };
 
-  extraConfig =
-    cfg:
-    mkMerge [
-      {
-        plugins.airline.settings.theme = mkIf cfg.setUpBar "base16";
-        plugins.lualine.theme = mkIf cfg.setUpBar "base16";
-        plugins.lightline.colorscheme = null;
-
-        opts.termguicolors = mkDefault true;
-      }
-      (mkIf (isString cfg.colorscheme) { colorscheme = "base16-${cfg.colorscheme}"; })
-      (mkIf (isAttrs cfg.colorscheme) {
-        extraConfigLuaPre = ''
-          require('base16-colorscheme').setup(${helpers.toLuaObject cfg.colorscheme})
-        '';
-      })
-    ];
+  # We will manually set the colorscheme, using `setup`
+  colorscheme = null;
+  callSetup = false;
+
+  extraConfig = cfg: {
+    plugins.airline.settings.theme = mkIf cfg.setUpBar (mkDefault name);
+    plugins.lualine.theme = mkIf cfg.setUpBar (mkDefault name);
+    plugins.lightline.colorscheme = mkDefault null;
+
+    opts.termguicolors = mkDefault true;
+    # FIXME: is it still ok to use `colorscheme` after using `setup`?
+    # It would be redundant, but maybe users want to read `config.colorscheme` in their nix config?
+    colorscheme = mkForce null;
+
+    # `settings` can either be passed to `with_config` before calling `setup`,
+    # or it can be passed as `setup`'s 2nd argument.
+    # See https://github.com/RRethy/base16-nvim/blob/6ac181b5733518040a33017dde654059cd771b7c/lua/base16-colorscheme.lua#L107-L125
+    extraConfigLuaPre = ''
+      require('base16-colorscheme').setup(${helpers.toLuaObject cfg.colorscheme}, ${helpers.toLuaObject cfg.settings})
```

This is also valid:
```suggestion
      require('base16-colorscheme').with_config(${helpers.toLuaObject cfg.settings})
      require('base16-colorscheme').setup(${helpers.toLuaObject cfg.colorscheme})
```
Which do you think is better?

---

I don't think it makes a huge difference, but having the `settings` option not feeding `require(..).setup` is quite weird. Pay attention to the description of the `settings` option itself.

---

> having the `settings` option not feeding `require(..).setup` is quite weird.

Yep. Their `setup` function _can_ take the config as a second argument, but they also provide `with_config` which is used separately. I.e. _even weirder_ than a plugin that has some alternative function to `setup`!

> Pay attention to the description of the `settings` option itself.

```
Options provided to the `require('${luaName}').setup` function.
```

You're right, that's an issue. I think it likely already affects other plugins though? Unless we have some way of overriding the description... Maybe your upcoming mk*Plugin refactor should take this into account?

---

> You're right, that's an issue. I think it likely already affects other plugins though? Unless we have some way of overriding the description... Maybe your upcoming mk*Plugin refactor should take this into account?

Yes, this might happen to be inaccurate for certain plugins.
We could add a `settingsDescription` argument that defaults to "... provided to the require blablabla".

---

This was resolved in #1758